Clarify some wasm terms in the introduction (#8115)
bytecodealliance/wasmtime: docs/introduction.md

# Introduction

[Wasmtime][github] is a standalone runtime for WebAssembly, WASI, and the
Component Model by the [Bytecode Alliance][BA].

[WebAssembly] (abbreviated Wasm) is a binary instruction format that is designed
to be a portable compilation target for programming languages. Wasm binaries
typically have a `.wasm` file extension. In this documentation, we'll also use
the textual representation of the binary files, which have a `.wat` file
extension.

[WASI] (the WebAssembly System Interface) defines interfaces that provide a

> secure and portable way to access several operating-system-like features such as
> filesystems, Berkely sockets, clocks, and random numbers.

Reviewer: FWIW, there are low-level sockets and higher-level APIs like the HTTP proxy world. Not really worth jumping into that level of detail here, so I think just saying "networking" is good.

```suggestion
secure and portable way to access several operating-system-like features such as
filesystems, networking, clocks, and random numbers.
```

[The Component Model] is a Wasm architecture that provides a binary format for
portable, cross-language composition. More specifically, it supports the use of

> interfaces via which components can communicate with each other. The main
> interfaces that the component model supports are those from WASI.

Reviewer: Small nitpick tweak to this last sentence:

```suggestion
interfaces via which components can communicate with each other. WASI
is defined in terms of component model interfaces.
```
